src/Symfony/Bundle/FrameworkBundle/Command/CacheClearCommand.php

```diff
@@ -104,34 +104,50 @@ protected function execute(InputInterface $input, OutputInterface $output)
             $io->comment('Clearing outdated warmup directory...');
         }
         $fs->remove($warmupDir);
-        $fs->mkdir($warmupDir);
 
-        if (!$input->getOption('no-warmup')) {
+        if ($_SERVER['REQUEST_TIME'] <= filemtime($containerFile) && filemtime($containerFile) <= time()) {
             if ($output->isVerbose()) {
-                $io->comment('Warming up cache...');
+                $io->comment('Cache is fresh.');
+            }
+            if (!$input->getOption('no-warmup') && !$input->getOption('no-optional-warmers')) {
+                if ($output->isVerbose()) {
+                    $io->comment('Warming up optional cache...');
+                }
+                $warmer = $kernel->getContainer()->get('cache_warmer');
+                // non optional warmers already ran during container compilation
+                $warmer->enableOnlyOptionalWarmers();
+                $warmer->warmUp($warmupDir);
```

Comment:
shouldn't we create the dir here ?

Reply:
or shouldn't we make it run optional warmers with the actual cache dir directly (which is not this variable), as we won't swap folders ?

Reply:
good catch, fixed